weblog/requesthandler.go

```diff
@@ -7,26 +7,29 @@ import (
 	"strconv"
 	"strings"
 
-	"github.com/deis/logger/syslogish"
+	"github.com/deis/logger/logs"
+	"github.com/deis/logger/storage"
 	"github.com/gorilla/mux"
 )
 
 type requestHandler struct {
-	syslogishServer *syslogish.Server
+	storageAdapter storage.Adapter
 }
 
-func newRequestHandler(syslogishServer *syslogish.Server) *requestHandler {
-	return &requestHandler{syslogishServer: syslogishServer}
+func newRequestHandler(storageAdapter storage.Adapter) *requestHandler {
```

Comment:
then, what @krancour applies, but only to the underlying implementation of the interface. don't pass pointers to the interface around, but make sure that the receivers of the implementing struct `func`s are pointers. example: https://play.golang.org/p/6w3gkz97Cz